source/transfer_list.rst

```diff
@@ -902,5 +903,134 @@ This manifest contains the SPMC attribute node consumed by the SPMD
      - hdr_size
      - Holds a SPMC manifest image in DT format.
 
+.. _dice_entries:
+
+Entries related to DICE
+^^^^^^^^^^^^^^^^^^^^^^^
+
+These entries hold [DICE]_ and [Open-DICE]_ information elements that needs to be transferred from one boot stage to the next in a *layered* DICE architecture, i.e., one that does not employ the DICE Protection Environment (DPE).
+
+This includes the Compound Device Identifier (CDI) used for attestation (and optional sealing), as well as the accumulated certificate chain.
+
+.. note::
+  As described in [DICE]_, CDIs are very sensitive data and are as long-lived as the layers they represent.
+  Each layer is therefore responsible for the proper handling of the CDIs it owns, as well as the derived cryptographic key material.
+  In particular, any given layer is expected to use and erase the CDI it received.
+  While the CDI is used, the layer needs the ability to prevent access to the CDI and disclosure of the value.
+
+The following entry types are currently defined:
+
+**CDI for attestation (XFERLIST_DICE_CDI_ATTEST)**
+
+This entry type holds the CDI described in [DICE]_.
+In [Open-DICE]_ it is referred to as "Attestation CDI".
+
+The CDI computed by the Sender's boot stage SHALL be added to the TL.
+The CDI received from the previous boot stage (i.e., DICE layer) MUST NOT be forwarded.
+
+.. _tab_dice_cdi_attest:
+.. list-table:: DICE attestation CDI
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x200**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size (in bytes) of the CDI
+
+   * - cdi_attest
```

Comment:
Would a "profile identifier" in the header do the trick?

Reply:
> Would a "profile identifier" in the header do the trick?
Maybe, if there was a doc that mapped each supported "profile identifier" to a binary data format.

Reply:
> > Would a "profile identifier" in the header do the trick?
> 
> Maybe, if there was a doc that mapped each supported "profile identifier" to a binary data format.

In [Open-DICE v2.5](https://pigweed.googlesource.com/open-dice/+/refs/heads/main/docs/specification.md), `OpenDiceInput` has a `profileName` claim for that purpose.

At least Android's Open DICE specialisations have a document describing profile-specific instantiation choices, which are mapped to a bunch of [profile identifier strings](https://pigweed.googlesource.com/open-dice/+/refs/heads/main/docs/android.md#Versions).

For TCG DICE the situation is much more foggy.  All of the baseline (i.e., non-DPE) specs allow ample freedom for implementation choices, but they don't have an explicit protocol indicator for specifying the specific instantiations.  The upcoming DPE spec does, but only concerning the DPE itself.

In summary, ISTM that a sensible way forward could be to drop TCG support and scope the TLs in this PR to Open DICE only, adding a "profile id" field to all the newly defined headers.

WDYT?

Reply:
Actually, on reading OpenDICE some more, I don't think it's important for receivers to understand the binary format of `cdi_attest`. They blindly use this as input to the KDF when generating the next layer's `cdi_attest`.

Also, `profileName` is part of the specified OpenDICE CDI certificate so I don't think there's anything additional to specify here.

I agree that TEs in this PR should be specific to OpenDICE. However, I'm reluctant to invent a "profile id" field with specific instantiation values if TCG don't already do this. I don't think we're likely to get lots of (or even any) other (non-DPE) profiles we need to support with this, right?

Reply:
> Actually, on reading OpenDICE some more, I don't think it's important for receivers to understand the binary format of `cdi_attest`. They blindly use this as input to the KDF when generating the next layer's `cdi_attest`.

Sorry, I wanted to reply specifically on that point but got lost in the profile rabbit hole...  As you say above, CDIs are just byte strings they don't need any special treatment.

> Also, `profileName` is part of the specified OpenDICE CDI certificate so I don't think there's anything additional to specify here.

For chaining to work all layers need to synchronise on the crypto primitives.  So, there must be an explicit way to communicate that information at handover points.  One option is to pass all the details in-band (hash, KDF, and digital signatures algorithms).  Another option is to use indirection via the profile concept.  What would be your choice?

> I agree that TEs in this PR should be specific to OpenDICE. However, I'm reluctant to invent a "profile id" field with specific instantiation values if TCG don't already do this.

If we only focus on OpenDICE we are not restricted in our choices.  We can deal with TCG DICE if and when needed, adding another TL if the ones already defined are not good enough.

> I don't think we're likely to get lots of (or even any) other (non-DPE) profiles we need to support with this, right?

I plan to define one such OpenDICE profile and use it, but effectively I don’t know of anyone else wanting to do the same, at least at this time.

Reply:
> For chaining to work all layers need to synchronise on the crypto primitives...

OK, then it does sound important for the next layer to understand the previous layer's `profileName` (before any certificates are constructed). I think it's enough to pass this info in the TEs rather than all the CDI details.

> I plan to define one such OpenDICE profile and use it,..

Sounds good.

Reply:
I gave it an initial stab at a89f7d4